examples/src/main/java/org/apache/ignite/examples/datagrid/CacheContinuousQueryWithTransformerExample.java

```diff
@@ -0,0 +1,151 @@
+ * Remote nodes should always be started with special configuration file which
+ * enables P2P class loading: {@code 'ignite.{sh|bat} examples/config/example-ignite.xml'}.
+ * <p>
+ * Alternatively you can run {@link ExampleNodeStartup} in another JVM which will
+ * start node with {@code examples/config/example-ignite.xml} configuration.
+ *
+ * @see CacheContinuousQueryExample
+ * @see CacheContinuousAsyncQueryExample
+ * @see ContinuousQueryWithTransformer
+ */
+public class CacheContinuousQueryWithTransformerExample {
+    /** Cache name. */
+    private static final String CACHE_NAME = CacheContinuousQueryExample.class.getSimpleName();
+
+    public static void main(String[] args) throws Exception {
+        try (Ignite ignite = Ignition.start("examples/config/example-ignite.xml")) {
+            System.out.println();
+            System.out.println(">>> Cache continuous query with transformer example started.");
+
+            // Auto-close cache at the end of the example.
+            try (IgniteCache<EmployeeKey, Employee> cache = ignite.getOrCreateCache(CACHE_NAME)) {
+                // Create new continuous query with transformer.
+                ContinuousQueryWithTransformer<EmployeeKey, Employee, String> qry =
+                    new ContinuousQueryWithTransformer<>();
+
+                //Factory to create transformers.
+                Factory<IgniteClosure<CacheEntryEvent<? extends EmployeeKey, ? extends Employee>, String>> factory =
+                    FactoryBuilder.factoryOf(
+                        //Return one field of complex object.
+                        //Only returned field will be send to local node over network.
+                        (IgniteClosure<CacheEntryEvent<? extends EmployeeKey, ? extends Employee>, String>)
+                            event -> event.getValue().name());
+
+                qry.setRemoteTransformerFactory(factory);
+
+                //Listener that will receive transformed data.
+                qry.setLocalListener(names -> {
+                    for (String name : names)
+                        System.out.println("New employee name: " + name);
+                });
+
+                // Execute query.
+                try (QueryCursor<Cache.Entry<EmployeeKey, Employee>> cur = cache.query(qry)) {
+                    populateCache(cache);
+
+                    // Wait for a while while callback is notified about remaining puts.
+                    Thread.sleep(2000);
+                }
+            }
+            finally {
+                // Distributed cache could be removed from cluster only by #destroyCache() call.
+                ignite.destroyCache(CACHE_NAME);
+            }
+        }
+    }
+
+    /**
+     * Populates cache with data.
+     *
+     * @param cache Employee cache.
+     */
+    @SuppressWarnings("TypeMayBeWeakened")
+    private static void populateCache(IgniteCache<EmployeeKey, Employee> cache) {
+        cache.put(new EmployeeKey(1, 1), new Employee(
```

Comment:
I would rather use 'putAll' instead of sequential puts just not to form a perception that it's a recommended way for data preloading. You can put all the entries into a HashSet and pass it into 'ignitecache.putAll'

Reply:
fixed